frontend/apps/reader/modules/readerwikipedia.lua

```diff
@@ -59,6 +61,146 @@ function ReaderWikipedia:addToMainMenu(menu_items)
             end
         end
     }
+    menu_items.wikipedia_settings = {
+        text = _("Wikipedia settings"),
+        sub_item_table = {
+            {
+                text = _("Set Wikipedia languages"),
+                callback = function()
+                    local wikilang_input
+                    local function save_wikilang()
+                        local wiki_languages = {}
+                        local langs = wikilang_input:getInputText()
+                        for lang in langs:gmatch("%S+") do
+                            if not lang:match("^[%a-]+$") then
+                                UIManager:show(InfoMessage:new{
+                                    text = T(_("%1 does not look like a valid Wikipedia language."), lang)
+                                })
+                                return
+                            end
+                            lang = lang:lower()
+                            table.insert(wiki_languages, lang)
+                        end
+                        G_reader_settings:saveSetting("wikipedia_languages", wiki_languages)
+                        -- re-init languages
+                        self.wiki_languages = {}
+                        self:initLanguages()
+                        UIManager:close(wikilang_input)
+                    end
+                    -- Use the list built by initLanguages (even if made from UI
+                    -- and document languages) as the initial value
+                    self:initLanguages()
+                    local curr_languages = table.concat(self.wiki_languages, " ")
+                    wikilang_input = InputDialog:new{
+                        title = _("Wikipedia languages"),
+                        input = curr_languages,
+                        input_hint = "en fr zh",
+                        input_type = "text",
+                        description = _("Enter one or more wikipedia language codes (the 2 or 3 letters before .wikipedia.org), in the order you wish to see them available, separated by space(s) (ie: en fr zh)\nFull list at https://en.wikipedia.org/wiki/List_of_Wikipedias"),
```

Comment:
* Missing capitalization in `wikipedia`
* `i.e.,` or just `example:` The fact that I use cf. and i.e. constantly just means I read a fair amount of academic material. :-P